composer_migration/src/composer_migration/lib/comparator.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, TypeVar
+import logging
+import os
+
+from composer_migration.lib.strategies.strategy import DAGChecker
+
+from rich.console import Console
+from rich.table import Table
+from rich.console import Group
+from rich.text import Text
+from rich.panel import Panel
+
+# https://peps.python.org/pep-0484/#annotating-instance-and-class-methods
+T = TypeVar("T", bound="DAGComparator")
+
+
+class DAGComparator:
+    def __init__(self: T, dags_directory: str) -> None:
+        self.problem_operators: List[str] = []
```

Comment:
is the type List[str] or a List[Tuple(str, str)] , where each tuple is (dag_file, operator) as per line #53  ?

Reply:
spot on, nothing gets past you!